File: leetcode/string/2243_calculate_digit_sum_of_a_string.ts
const digitSum = (s: string, k: number): string => {
    while (s.length > k) {
        let _sum = "";
        const len = s.length;
        for (let i = 0; i < len; i += k) {
            let sum = 0;
            const last = i + k > len ? len : i + k;
            for (let j = i; j < last; j++) {
                sum += parseInt(s[j]);
            }
            _sum += sum;
        }
        s = _sum;
        console.log(s);
    }
    return s;
};

File: string/2243_calculate_digit_sum_of_a_string.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    string digitSum(string s, int k)
    {
        while (s.length() > k)
        {
            int len = s.length();
            string _sum = "";
            for (int i = 0; i < len; i += k)
            {
                int sum = 0;
                int last = i + k > len ? len : i + k;
                for (int j = i; j < last; j++)
                {
                    sum += (s[j] - '0');
                }
                _sum += to_string(sum);
            }
            s = _sum;
            cout << s << "\n";
        }
        return s;
    }
};

int main()
{
}
// by Shihab Mahamud (github.com/shh26b)
// Sunday, April 17, 2022 | 02:50:12 PM (+06)
